packages/browser/src/plugins/BrowserLifeCyclePlugin.ts

```diff
@@ -17,12 +17,19 @@ export class BrowserLifeCyclePlugin implements IEventPlugin {
 
     this._client = context.client;
 
-    globalThis.addEventListener("beforeunload", () => void this._client?.suspend());
+    globalThis.addEventListener("beforeunload", () => {
+      if (this._client?.config.sessionsEnabled) {
+        void this._client?.submitSessionEnd();
+      }
+
+      void this._client?.suspend();
```

Comment:
Is suspend what we want to do here? Wouldn't we want to flush the queue? Does suspend do that?

Reply:
```
    await EventPluginManager.suspend(new PluginContext(this));
    const { queue } = this.config.services;
    await queue.suspend();
    await queue.process();
    this.suspendSettingsTimer();
```

It shuts everything down (all timers and intervals) and then submits events.